server/auth/store.go

```diff
@@ -624,6 +627,10 @@ func (as *authStore) UserRevokeRole(r *pb.AuthUserRevokeRoleRequest) (*pb.AuthUs
 }
 
 func (as *authStore) RoleGet(r *pb.AuthRoleGetRequest) (*pb.AuthRoleGetResponse, error) {
+	if rootRole == r.Role {
```

Comment:
I feel this approach is a little bit risky because the result returned by `RoleGet()` might be used by other components of etcd server (it should be avoided of course). Also it changes the meaning of the fields. Although v2 employs the similar approach, I prefer the initial approach which only changes etcdctl. Note that v2's auth is based on path and v3's auth is based on prefix, so they are quite different.
How do you think @tangcong ?

Reply:
Actually `*` is a valid etcd key name.

Reply:
sounds reasonable. we need to be consistent with the existing rules, so we cannot use * on the server-side.  How about we use noPrefixEnd?  If this PR only changes etcdctl, it looks a little tricky, and there are other languages/sdk that will be called through API.